src/bin.js

```diff
@@ -2,4 +2,9 @@
 
 'use strict'
 
-require('./')()
+// First param is the target version
+// Second param is the target platform
+// Third param is the target architecture
+require('./')(process.argv[2], process.argv[3], process.argv[4])
+  .then(() => process.exit(0))
+  .catch((e) => process.exit(1))
```

Comment:
But you're right, the actual download function doesn't need to handle the output and it can be moved to bin.js (cli). Makes more sense in terms of responsibilities. Will change that.